src/client/components/pages/TestsHistory/TestsHistory.jsx

```diff
@@ -6,41 +6,54 @@ import {
   FaDownload as DownloadIcon,
 } from 'react-icons/fa';
 import Table from '../../organisms/Table/Table';
-import ExamCard from '../../organisms/ExamCard/ExamCard';
+import TestCard from '../../organisms/TestCard/TestCard';
 import Button from '../../atoms/Button/Button';
 import Checkbox from '../../atoms/Checkbox/Checkbox';
 import MainViewProvider from '../../providers/MainViewProvider/MainViewProvider';
 import NavbarProvider from '../../providers/NavbarProvider/NavbarProvider';
 import FeedbackProvider from '../../providers/FeedbackProvider/FeedbackProvider';
-import { listExams } from '../../../redux/exams/exams.actions.requests';
-import { setSelectedExams } from '../../../redux/exams/exams.actions';
-
-import { getStringFromDate } from '../../../utils/date';
-import { templateEditor } from '../../../routes/paths';
-
-const ExamsHistory = (props) => {
-  const { history } = props;
+import { getUser } from '../../../redux/user/user.actions.requests';
+import { setSelectedTests } from '../../../redux/user/user.actions';
 
+const TestsHistory = () => {
   const dispatch = useDispatch();
-  const { exams, selectedExams } = useSelector((state) => state.exams);
+  const { user, selectedTests } = useSelector((state) => state.user);
   const { isLoading } = useSelector((state) => state.feedback);
 
+  const [filteredTests, setFilteredTests] = useState(user ? user.tests : []);
+
   const handleCheckboxOnChange = (id) => (event) => {
     const { checked } = event.target;
     if (checked) {
-      dispatch(setSelectedExams({ selectedExams: [...selectedExams, id] }));
+      dispatch(setSelectedTests({ selectedTests: [...selectedTests, id] }));
     } else {
-      dispatch(setSelectedExams({
-        selectedExams: selectedExams.filter((examId) => examId !== id),
+      dispatch(setSelectedTests({
+        selectedTests: selectedTests.filter((testId) => testId !== id),
+      }));
+    }
+  };
+
+  const handleOnSearch = (query) => {
+    if (query) {
+      setFilteredTests(filteredTests.filter(({ name }) => {
+        return name.toUpperCase().includes(query.toUpperCase());
       }));
+    } else {
+      setFilteredTests(user.tests);
     }
   };
 
+  const editTe = () => {};
+
   useEffect(() => {
-    dispatch(listExams());
+    dispatch(getUser('5ec5f50ea76d8d1b91de94e5'));
```

Comment:
This is the burnt text?

Reply:
Hahahaah sorry, it  must be the hour. I'll fix that.

Reply:
Done :grin: